src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -61,6 +61,10 @@ function warnIfInvalidElement(Component, element) {
   }
 }
 
+function isNewable(Component) {
```

Comment:
`shouldConstruct` sounds good. A little misleading that we still call `_constructComponent` even though it can be `false` but I’m out of good function names at this point. (`instantiate*` is taken for another purpose :smile: )